File: Java Applet Examples/src/net/ptidej/teamvm/applet/MathApplet.js
<script src=
"https://www.java.com/js/deployJava.js"></script>
<script>
var attributes = { id:'mathApplet', code:'net.ptidej.teavm.MathApplet', ...};
var parameters = { jnlp_href: 'math_applet.jnlp'} ;
deployJava.runApplet(attributes, parameters, '1.6');
</script>

<script language="javascript">
    function enterNums(){
        var numA = prompt('Enter number \'a\'?','0');
        var numB = prompt(
            'Enter number \'b\' (should be greater than number \'a\' ?','1');
        // set applet's public variable
        mathApplet.userName = "John Doe";

        // invoke public applet method
        var greeting = mathApplet.getGreeting();

        // get another class referenced by applet and
        // invoke its methods
        var calculator = mathApplet.getCalculator();
        calculator.setNums(numA, numB);

        // primitive datatype returned by applet
        var sum = calculator.add();

        // array returned by applet
        var numRange = calculator.getNumInRange();

        // check Java console log for this message
        mathApplet.printOut("Testing printing to System.out");

        // get another class, set static field and invoke its methods
        var dateHelper = mathApplet.getDateHelper();
        dateHelper.label = "Today\'s date is: ";
        var dateStr = dateHelper.getDate();
        <!-- ... -->
</script>

File: Java Applet Examples/src/net/ptidej/teamvm/applet/MathApplet.java
// See https://docs.oracle.com/javase/tutorial/deployment/applet/invokingAppletMethodsFromJavaScript.html
package net.ptidej.teamvm.applet;

import java.applet.Applet;
import java.text.SimpleDateFormat;
import java.util.Date;

public class MathApplet extends Applet {
	private static final long serialVersionUID = 7204571981749946950L;

	public String userName = null;

	public String getGreeting() {
		return "Hello " + userName;
	}

	public Calculator getCalculator() {
		return new Calculator();
	}

	public DateHelper getDateHelper() {
		return new DateHelper();
	}

	public void printOut(String text) {
		System.out.println(text);
	}
}

class Calculator {
	private int a = 0;
	private int b = 0; // assume b > a

	public void setNums(int numA, int numB) {
		a = numA;
		b = numB;
	}

	public int add() {
		return a + b;
	}

	public int[] getNumInRange() {
		int x = a;
		int len = (b - a) + 1;
		int[] range = new int[len];
		for (int i = 0; i < len; i++) {
			range[i] = x++;
			System.out.println("i: " + i + " ; range[i]: " + range[i]);
		}
		return range;
	}
}

class DateHelper {
	public static String label = null;

	public String getDate() {
		return label + " " + new SimpleDateFormat().format(new Date());
	}

}